File: app/src/main/java/bzh/naoned/naonedbusdatadisplay/view/adapter/TargetAdapter.java
package bzh.naoned.naonedbusdatadisplay.view.adapter;

import android.databinding.DataBindingUtil;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import java.util.List;

import bzh.naoned.naonedbusdatadisplay.R;
import bzh.naoned.naonedbusdatadisplay.databinding.TargetListItemBinding;
import bzh.naoned.naonedbusdatadisplay.service.model.Target;

public class TargetAdapter extends RecyclerView.Adapter<TargetAdapter.TargetViewHolder> {
    private List<Target> targetsList;

    @NonNull
    @Override
    public TargetViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        TargetListItemBinding binding = DataBindingUtil
                .inflate(LayoutInflater.from(viewGroup.getContext()), R.layout.target_list_item, viewGroup, false);

        return new TargetViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull TargetViewHolder holder, int i) {
        holder.binding.setTarget(targetsList.get(i));
        holder.binding.executePendingBindings();
    }

    @Override
    public int getItemCount() {
        return targetsList == null ? 0 : targetsList.size();
    }

    public void setValues(List<Target> targets) {
        targetsList = targets;
        notifyItemRangeInserted(0, targets.size()
        );
    }

    static class TargetViewHolder extends RecyclerView.ViewHolder {

        final TargetListItemBinding binding;

        public TargetViewHolder(TargetListItemBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: app/src/main/java/bzh/naoned/naonedbusdatadisplay/view/adapter/TargetAdapter.kt
package bzh.naoned.naonedbusdatadisplay.view.adapter

import android.databinding.DataBindingUtil
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup

import bzh.naoned.naonedbusdatadisplay.R
import bzh.naoned.naonedbusdatadisplay.databinding.TargetListItemBinding
import bzh.naoned.naonedbusdatadisplay.service.model.Target

internal class TargetAdapter : RecyclerView.Adapter<TargetAdapter.TargetViewHolder>() {
    private var targetsList: List<Target>? = null

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): TargetViewHolder {
        val binding = DataBindingUtil
                .inflate<TargetListItemBinding>(LayoutInflater.from(viewGroup.context), R.layout.target_list_item, viewGroup, false)

        return TargetViewHolder(binding)
    }

    override fun onBindViewHolder(holder: TargetViewHolder, i: Int) {
        holder.binding.setTarget(targetsList?.get(i))
        holder.binding.executePendingBindings()
    }

    override fun getItemCount(): Int {
        return targetsList?.size?:0
    }

    fun setValues(targets: List<Target>?) {
        targetsList = targets
        notifyItemRangeInserted(0, targets?.size?:0)
    }

    internal class TargetViewHolder(val binding: TargetListItemBinding) : RecyclerView.ViewHolder(binding.getRoot())
}
